pkg/files/filehandling.go

```diff
@@ -46,8 +50,39 @@ func setDefaultConfig() {
 
 // InitFileHandler creates a new file handler for the file backend we want to use
 func InitFileHandler() {
-	fs = afero.NewOsFs()
-	afs = &afero.Afero{Fs: fs}
+	fileType := config.FilesType.GetString()
+	
+	if fileType == "s3" {
+		// Get S3 configuration
+		endpoint := config.FilesS3Endpoint.GetString()
+		bucket := config.FilesS3Bucket.GetString()
+		region := config.FilesS3Region.GetString()
+		accessKey := config.FilesS3AccessKey.GetString()
+		secretKey := config.FilesS3SecretKey.GetString()
+
+		if endpoint == "" || bucket == "" || accessKey == "" || secretKey == "" {
+			log.Fatal("S3 configuration incomplete. Please set files.s3.endpoint, files.s3.bucket, files.s3.accesskey, and files.s3.secretkey")
+		}
+
+		// Create AWS session for afero-s3
+		sess, err := session.NewSession(&aws.Config{
+			Region:      aws.String(region),
+			Credentials: credentials.NewStaticCredentials(accessKey, secretKey, ""),
+			Endpoint:    aws.String(endpoint),
+			S3ForcePathStyle: aws.Bool(true), // Required for some S3-compatible services
```

Comment:
Why is this necessary? Should it be user-configurable?

Reply:
@kolaente (not the author) the force path style is commonly needed by selfhosters because serving buckets using path-style is really easier. This parameter is often overlooked

Reply:
Fine by me, but should it be configurable?

Reply:
Yes it should as some S3 services do only dns-style, some do only path-style.
I would probably change the the name to S3**Use**PathStyle instad of S3**Force**PathStyle like they [did in pulumi](https://github.com/pulumi/pulumi-aws/issues/3068) so that it doesn't look like one is forcing something bad, but that's nitpicking.